src/videostreamindex/DefaultVideoStreamIndex.ts

```diff
@@ -277,24 +281,16 @@ export default class DefaultVideoStreamIndex implements VideoStreamIndex {
 
   streamIdForTrack(trackId: string): number {
```

Comment:
All other getter APIs, modified in this PR, are called after `SubscribeAndReceiveSubscribeAckTask` completes. For example, after the "update" operation, the RTCPeerConnection track handler is calling `attendeeIdForTrack` to get the attendee ID. PR makes this attendee ID valid until the next subscribe ack.

But `streamIdForTrack` can be called before `SubscribeAndReceiveSubscribeAckTask`. Here's a stack trace:

1. The "update" operation always executes `AttachMediaInputTask` before `SubscribeAndReceiveSubscribeAckTask`.
2. `AttachMediaInputTask` calls `transceiverController.updateVideoTransceivers`.
3. `transceiverController.updateVideoTransceivers` calls `transceiverController.unsubscribeTransceivers`.
3. `transceiverController.unsubscribeTransceivers` calls this `streamIdForTrack` API.
4. After `AttachMediaInputTask` completes, `SubscribeAndReceiveSubscribeAckTask`

Could you review this flow too when SDK receives multiple video streams?

Reply:
Please ignore this comment. The consecutive "update" operation will be deferred, so each "udpate" operation unsubscribes transceivers using the previously-built subscribe map. I can't think of race conditions as the map will be updated per `SubscribeAndReceiveSubscribeAckTask`.

![image](https://user-images.githubusercontent.com/36240708/91901518-30017a80-ec55-11ea-82f4-a59a0d5c01ef.png)